File: src/main/java/com/yzh/campushub/entity/Post.java
package com.yzh.campushub.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("post")
public class Post implements Serializable {
    private static final long serialVersionUID = 1L;
    @TableId(value = "id",type = IdType.AUTO)
    private Long id;

    private Long userId;
    private Long categoryId;
    private String title;
    private String content;
    private String coverImg;
    private Integer viewCount;
    private Integer likeCount;
    private Integer commentCount;
    private Integer favoriteCount;
    private Integer status;
    private Integer isTop;
    private Integer isDeleted;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;

}

File: frontend/src/types/post.ts
export interface PostQueryDTO {
  pageNum: number
  pageSize: number
  categoryId?: number
  sortType?: 'latest' | 'hottest'
  keyword?: string
}

export interface PostVO {
  id: number
  userId: number
  nickname: string
  avatar?: string
  categoryId: number
  categoryName?: string
  title: string
  content: string
  coverImg?: string
  viewCount: number
  likeCount: number
  commentCount: number
  favoriteCount: number
  status?: number
  isTop?: number
  createTime?: string
  updateTime?: string
}

export interface PostAuthorVO {
  id: number
  nickname: string
  avatar?: string
}

export interface PostDetailVO {
  id: number
  userId?: number
  categoryId?: number
  title: string
  content: string
  coverImg?: string
  viewCount: number
  likeCount: number
  commentCount: number
  favoriteCount: number
  createTime?: string
  updateTime?: string
  images: string[]
  author: PostAuthorVO
  isLiked: boolean
  isFavorited: boolean
}

export interface SavePostDTO {
  categoryId: number
  title: string
  content: string
  images: string[]
}
